src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -721,11 +721,39 @@ InvokeHostFunctionOpFrame::doCheckValid(uint32_t ledgerVersion)
     throw std::runtime_error(
         "InvokeHostFunctionOpFrame::doCheckValid needs Config");
 }
-
 void
 InvokeHostFunctionOpFrame::insertLedgerKeysToPrefetch(
     UnorderedSet<LedgerKey>& keys) const
 {
+    UnorderedMap<LedgerKey, UnorderedSet<Hash>> lkToTx;
+    insertLedgerKeysWithMappingsToPrefetch(keys, lkToTx);
+}
+void
+InvokeHostFunctionOpFrame::insertLedgerKeysWithMappingsToPrefetch(
```

Comment:
This logic should be common for all Soroban ops. Moreover, it really belongs to the transaction layer, as resources are defined at transaction level (not operation level).